v2/api/dataprotection/v1api20230101/structure.txt

```diff
@@ -0,0 +1,1272 @@
+│   │   ├── CreatedAt: *string
+│   │   ├── CreatedBy: *string
+│   │   ├── CreatedByType: *Enum (4 values)
+│   │   │   ├── "Application"
+│   │   │   ├── "Key"
+│   │   │   ├── "ManagedIdentity"
+│   │   │   └── "User"
+│   │   ├── LastModifiedAt: *string
+│   │   ├── LastModifiedBy: *string
+│   │   └── LastModifiedByType: *Enum (4 values)
+│   │       ├── "Application"
+│   │       ├── "Key"
+│   │       ├── "ManagedIdentity"
+│   │       └── "User"
+│   ├── Tags: map[string]string
+│   └── Type: *string
+├── BackupVault_Spec_ARM: Object (6 properties)
+│   ├── ETag: *string
+│   ├── Identity: *Object (1 property)
+│   │   └── Type: *string
+│   ├── Location: *string
+│   ├── Name: string
+│   ├── Properties: *Object (4 properties)
+│   │   ├── FeatureSettings: *Object (1 property)
+│   │   │   └── CrossSubscriptionRestoreSettings: *Object (1 property)
+│   │   │       └── State: *Enum (3 values)
+│   │   │           ├── "Disabled"
+│   │   │           ├── "Enabled"
+│   │   │           └── "PermanentlyDisabled"
+│   │   ├── MonitoringSettings: *Object (1 property)
+│   │   │   └── AzureMonitorAlertSettings: *Object (1 property)
+│   │   │       └── AlertsForAllJobFailures: *Enum (2 values)
+│   │   │           ├── "Disabled"
+│   │   │           └── "Enabled"
+│   │   ├── SecuritySettings: *Object (2 properties)
+│   │   │   ├── ImmutabilitySettings: *Object (1 property)
+│   │   │   │   └── State: *Enum (3 values)
+│   │   │   │       ├── "Disabled"
+│   │   │   │       ├── "Locked"
+│   │   │   │       └── "Unlocked"
+│   │   │   └── SoftDeleteSettings: *Object (2 properties)
+│   │   │       ├── RetentionDurationInDays: *float64
+│   │   │       └── State: *Enum (3 values)
+│   │   │           ├── "AlwaysOn"
+│   │   │           ├── "Off"
+│   │   │           └── "On"
+│   │   └── StorageSettings: Object (2 properties)[]
+│   │       ├── DatastoreType: *Enum (3 values)
+│   │       │   ├── "ArchiveStore"
+│   │       │   ├── "OperationalStore"
+│   │       │   └── "VaultStore"
+│   │       └── Type: *Enum (3 values)
+│   │           ├── "GeoRedundant"
+│   │           ├── "LocallyRedundant"
+│   │           └── "ZoneRedundant"
+│   └── Tags: map[string]string
+├── BackupVaultsBackupInstance: Resource
+│   ├── Owner: BackupVault
+│   ├── Spec: Object (5 properties)
+│   │   ├── AzureName: string
+│   │   ├── Owner: *genruntime.KnownResourceReference
+│   │   ├── Properties: *Object (7 properties)
+│   │   │   ├── DataSourceInfo: *Object (7 properties)
+│   │   │   │   ├── DatasourceType: *string
+│   │   │   │   ├── ObjectType: *string
+│   │   │   │   ├── ResourceLocation: *string
+│   │   │   │   ├── ResourceName: *string
+│   │   │   │   ├── ResourceReference: *genruntime.ResourceReference
+│   │   │   │   ├── ResourceType: *string
+│   │   │   │   └── ResourceUri: *string
+│   │   │   ├── DataSourceSetInfo: *Object (7 properties)
+│   │   │   │   ├── DatasourceType: *string
+│   │   │   │   ├── ObjectType: *string
```

Comment:
Same question here, how is this used in practice? (Maybe I will see in the samples/tests)

Reply:
Here also, ResourceID is a required ALWAYS field, and the rest of the other fields, including ResourceLocation, ResourceName, and ResourceType, are required default.